client/components/jetpack/upsell-switch/index.tsx

```diff
@@ -34,53 +35,66 @@ type Props = {
 	display: ReactElement;
 	QueryComponent: ComponentType< QueryComponentProps >;
 	getStateForSite: QueryFunction;
+	isRequestingForSite: RequestFunction;
 	siteId: number | null;
 	siteState: SiteState | null;
 	atomicSite: boolean;
+	isRequesting: boolean;
 };
 
+const UI_STATE_LOADING = Symbol();
+const UI_STATE_LOADED = Symbol();
```

Comment:
This is a really cool use case for `Symbol`! One thing I'd wonder about is the overhead of transpiling to browsers that don't support it. It'd be easy to substitute strings, but not a huge deal either way.

Reply:
True! I've seen it used in this way in other parts of the codebase but the polyfill won't work for more advanced features of Symbol indeed.